### PR TITLE
scheduler: remove scaler, GCE, OpenStack, and manual

### PR DESCRIPTION
**Pull request by buchanae**
For the reasons listed in #325, I think it's best we remove the autoscaling code for now.

This PR removes over 1.5K lines of code that was mostly unused, broken, and getting in the way of features we're actively developing such as event writers, pluggable components, end-to-end encryption and security, and configuration.

This also substantially simplifies the scheduler, since it doesn't need multiple backends now.

### CODE REVIEW

**Comment by adamstruck** [comment]
The example config and website need to be updated.

File: config/default-config.yaml [hunk 1]
```diff
@@ -61,7 +60,7 @@ Server:
     # Write logs to this path. If empty, logs are written to stderr.
     OutputFile: ""
 
-# The scheduler is used for the GCE, Openstack, and Manual compute backends. 
+# The scheduler is used for the Openstack, and Manual compute backends.
```
Comment (adamstruck) [inline config/default-config.yaml]: There is no openstack backend anymore

File: config/default-config.yaml [hunk 2]
```diff
@@ -112,7 +111,7 @@ Scheduler:
 
 
 # The name of the active compute backend
-# Available backends: local, htcondor, slurm, pbs, gridengine, manual, gce, openstack, aws-batch
+# Available backends: local, htcondor, slurm, pbs, gridengine, manual, openstack, aws-batch
```
Comment (adamstruck) [inline config/default-config.yaml]: There is no openstack backend anymore

**Comment by adamstruck** [comment]
Did you test that the GCE deployment tools still work?

**Comment by buchanae** [comment]
I also forgot to build the examples bundle. Updated.

**Pull request merged**